repo_resource/common.py

```diff
@@ -253,6 +254,24 @@ def __get_remote_url(self, remote):
     def __get_remote_revision(self, remote):
         return self.__remote_revision[remote]
 
+    def set_insteadOf(self, matrix: dict = None):
+        with redirect_stdout(sys.stderr):
+            if matrix is not None:
+                tempdir = tempfile.TemporaryDirectory(delete=False)
```

Comment:
Can we use this as a context manager instead?
```python
with tempfile.TemporaryDirectory() as tmpdirname:
```

This is how it's used elsewhere and will perform cleanup even if gitrepo.config_write() raises an exception


See: https://docs.python.org/3/library/tempfile.html#examples